runtime/runtime/src/lib.rs

```diff
@@ -122,6 +122,8 @@ pub struct ApplyResult {
     pub state_changes: Vec<RawStateChangesWithTrieKey>,
     pub stats: ApplyStats,
     pub proof: Option<PartialStorage>,
+    /// Receipts restored during runtime apply after migration.
+    pub receipts_to_restore: Vec<Receipt>,
```

Comment:
Why do we actually need this field? I thought we have receipts to restore as a global variable

Reply:
We do, but we need `chain_id` to decide to load it or not, which is stored on `Runtime` side.
And also at some moment in the past we decided that `Chain` should not know about `chain_id`, which discourages adding method like `get_chain_id` to `RuntimeAdapter`. So I leave `receipts_to_restore` computation to `Runtime`.

Reply:
This is not a great solution. If possible, we should avoid introducing some field to `ApplyResult` only to use it once.

> also at some moment in the past we decided that Chain should not know about chain_id

I think it is more like we didn't see a good reason to do that. It does not mean it should never be done. Alternatively we can update it in `process_state_update` by writing directly to the database, which is also an ugly solution.